File: packages/dataframe/rust/stats/distributions/uniform.test.R
#!/usr/bin/env Rscript

# Uniform distribution test script
# Usage: Rscript uniform.test.R <function> <x_or_p> <min> <max> [lower_tail] [log_p]

source("test-helpers.R")

# Define uniform distribution functions
uniform_density <- function(x, min, max, log = FALSE) {
  dunif(x, min, max, log = log)
}

uniform_cumulative <- function(x, min, max, lower.tail = TRUE, log.p = FALSE) {
  punif(x, min, max, lower.tail = lower.tail, log.p = log.p)
}

uniform_quantile <- function(p, min, max, lower.tail = TRUE, log.p = FALSE) {
  qunif(p, min, max, lower.tail = lower.tail, log.p = log.p)
}

uniform_random <- function(n, min, max) {
  runif(n, min, max)
}

# Run the test
run_distribution_test(uniform_density, uniform_cumulative, uniform_quantile, uniform_random)

File: packages/dataframe/rust/stats/distributions/uniform.test.ts
#!/usr/bin/env -S deno run --allow-read --allow-run

import { callR, printResults, runComparison } from "./test-helpers.ts";

async function main() {
  console.log("Testing R vs Rust uniform functions...\n");

  const testCases = [
    { func: "dunif", distribution: "uniform", args: [0.5, 0, 1, 0] },
    { func: "punif", distribution: "uniform", args: [0.5, 0, 1, 1, 0] },
    { func: "qunif", distribution: "uniform", args: [0.5, 0, 1, 1, 0] },
  ];

  const results = await runComparison(testCases);
  printResults(results);

  // Test runif separately since it's not available in Rust interface
  console.log("\nR-only tests:");
  try {
    const runifResult = await callR("runif", "uniform", 1, 0, 1);
    console.log(`runif(0, 1) = ${runifResult}`);
  } catch (error) {
    console.log(`runif: Error - ${(error as Error).message}`);
  }
}

if (import.meta.main) {
  main().catch(console.error);
}
